File: src/main/java/com/coding/programmers/level2/구명보트.java
package com.coding.programmers.level2;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.Queue;

public class 구명보트 {
    public static void main(String[] args) {
        Solution sol = new Solution();
        System.out.println(sol.solution(new int[]{70,50,80,50}, 100));
    }
    private static class Solution {
        public int solution(int[] people, int limit) {
            int answer = 0;
            Arrays.sort(people);
            int front = 0;
            int rear = people.length - 1;

            while (front <= rear) {
                if ((people[front] + people[rear]) <= limit) {
                    front++;
                    rear--;
                } else {
                    rear--;
                }
                answer++;
            }

            return answer;
        }
    }
}

File: src/main/kotlin/com/coding/programmers/greedy/구명보트.kt
package com.coding.programmers.greedy

fun main() {
    val result = 구명보트().solution(intArrayOf(70, 50, 80, 50),100)
    println(result)
    val result2 = 구명보트().solution(intArrayOf(70, 80, 50),100)
    println(result2)
}

class 구명보트 {
    fun solution(people: IntArray, limit: Int): Int {
        var answer = 0
        val p = people.sortedArray()
        var left = 0
        var right = p.size - 1
        while (left <= right) {
            if (p[left] + p[right] <= limit) {
                left++
            }
            right--
            answer++
        }

        return answer
    }
}